### ISSUE TITLE
osc over Ethernet is not stable...

### ISSUE THREAD

**Issue by rgetz** [analogdevicesinc/iio-oscilloscope#245]
periodically, the capture will freeze.

In the console, we see...

```
FPS: 9
FPS: 8
FPS: 8
FPS: 10
FPS: 7
FPS: 8
Error while reading data: (32) Broken pipe
```

clicking on stop/start will re-start things.

**Comment by rgetz** [comment]
I tried with something like this:

```
diff --git a/osc.c b/osc.c
index d07a465..844b43c 100644
--- a/osc.c
+++ b/osc.c
@@ -1241,8 +1241,13 @@ static gboolean capture_process(void *data)
                while (true) {
                        ssize_t ret = iio_buffer_refill(dev_info->buffer);
                        if (ret < 0) {
-                               fprintf(stderr, "Error while reading data: %s\n", strerror(-ret));
+                               fprintf(stderr, "Error while reading data: (%zi) %s\n", -ret, strerror(-ret));
                                stop_sampling();
+                               if (ret == -EPIPE ) {
+                                       stop_capture = FALSE;
+                                       capture_setup();
+                                       break;
+                               }
                                goto capture_stop_check;
                        }
 
```

and that mostly fixes things - but it still fails sometimes...

**Comment by rgetz** [comment]
closed with #248
